src/data/repositories/DataSetD2Api.ts

```diff
@@ -1,91 +1,102 @@
-                            "attributeValues.value": this.isProjectFilterPresent(options)
-                                ? { in: [...this.getOrThrow(options.filters.projectsIds)] }
-                                : { eq: "true" },
-                            identifiable: { token: options.filters.search },
-                            id: { in: options.filters.ids },
-                        },
-                        fields: {
-                            ...dataSetFields,
-                            ...(options.includeOrgUnits
-                                ? {
-                                      organisationUnits: {
-                                          id: true,
-                                          path: true,
-                                          displayName: true,
-                                      },
-                                  }
-                                : {}),
-                        },
-                        order: `${options.sorting.field}:${options.sorting.order}`,
-                    })
-                ).flatMap(d2Response => {
-                    const projectIds = this.getProjectIds(d2Response.objects);
-                    return this.getProjectsByIds(projectIds).map(projects => {
-                        const dataSets = d2Response.objects.map(d2DataSet => {
-                            return this.buildDataSet(d2DataSet, coreCompetencies, projects);
-                        });
-                        return { ...d2Response.pager, data: dataSets };
+        return this.getBaseData().flatMap(({ attributes, coreCompetencies }) => {
+            const attributesToFilter = options.filters.projectsIds
+                ? [attributes.project.id]
+                : [attributes.createdByApp.id];
+            return apiToFuture(
+                this.api.models.dataSets.get({
+                    pageSize: options.paging.pageSize,
+                    page: options.paging.page,
+                    filter: {
+                        "attributeValues.attribute.id": { in: attributesToFilter },
+                        "attributeValues.value": this.isProjectFilterPresent(options)
+                            ? { in: [...this.getOrThrow(options.filters.projectsIds)] }
+                            : { eq: "true" },
+                        identifiable: { token: options.filters.search },
+                        id: { in: options.filters.ids },
+                    },
+                    fields: {
+                        ...dataSetFields,
+                        ...(options.includeOrgUnits
+                            ? {
+                                  organisationUnits: {
+                                      id: true,
+                                      path: true,
+                                      displayName: true,
+                                  },
+                              }
+                            : {}),
+                    },
```

Comment:
Conditional fields will bring problems. If you check the response type, organisationUnits won't be nice to work with. It can be done with generics, but typically is better to keep it simple. If we need a method that does not get orgunits, then we have a separate method and a separate dataSet type (without orgUnit).